code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -77,8 +79,9 @@
 
 /obj/effect/particle_effect/smoke/Crossed(atom/movable/O)
 	. = ..()
-	if(CHECK_BITFIELD(smoke_traits, SMOKE_CAMO) && isliving(O))
+	if(isliving(O))
 		O.effect_smoke(src)
+		return
 	if(CHECK_BITFIELD(smoke_traits, SMOKE_NERF_BEAM) && istype(O, /obj/projectile))
```

Comment:
Is this still needed?

Reply:
It's kind of a niche case, but yeah. Shooting energy weapons, including lasers, though certain smoke clouds cuts their damage. M40 HSDPs use that smoke, don't think anything else does.